src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -838,8 +841,13 @@ private CodeTypeBase CreateModelDeclarations(OpenApiUrlTreeNode currentNode, Ope
         else throw new InvalidOperationException("un handled case, might be object type or array type");
     }
     private CodeElement GetExistingDeclaration(bool checkInAllNamespaces, CodeNamespace currentNamespace, OpenApiUrlTreeNode currentNode, string declarationName) {
-        var searchNameSpace = GetSearchNamespace(checkInAllNamespaces, currentNode, currentNamespace);
-        return searchNameSpace.FindChildByName<ITypeDefinition>(declarationName, checkInAllNamespaces) as CodeElement;
+        var localNameSpace = GetSearchNamespace(false, currentNode, currentNamespace);
```

Comment:
This change involves first searching for the type in the target namespace before checking the root namespace. This fixes incorrect type matching observed when looking for `microsoft.graph.termstore.group` incorrectly finding `microsoft.graph.group`

Reply:
thanks for the detailed explanation